pyatlan/model/fields/atlan_fields.py

```diff
@@ -0,0 +1,528 @@
+from abc import ABC
+from typing import Union
+
+from pydantic import StrictBool, StrictFloat, StrictInt, StrictStr
+
+from pyatlan.model.enums import SortOrder
+from pyatlan.model.search import (
+    Exists,
+    Match,
+    Prefix,
+    Query,
+    Range,
+    SearchFieldType,
+    SortItem,
+    Term,
+    Terms,
+)
+from pyatlan.model.typedef import AttributeDef
+
+
+class AtlanField(ABC):
+    """
+    Base enumeration of all attributes that exist in Atlan, so you do not have to remember their
+    exact spelling or capitalization.
+    """
+
+    atlan_field_name: StrictStr
+
+
+class RelationField(AtlanField):
+    """
+    Represents any field used to capture a relationship in Atlan, which is not inherently
+    searchable.
+    """
+
+    def __init__(self, atlan: StrictStr):
+        """
+        Default constructor.
+
+        :param atlan: name of the attribute in the metastore
+        """
+        self.atlan_field_name = atlan
+
+
+class SearchableField(AtlanField):
+    """
+    Base class for any field in Atlan that can be searched.
+    """
+
+    elastic_field_name: StrictStr
+
+    def __init__(self, atlan: StrictStr, elastic: StrictStr):
+        """
+        Default constructor.
+
+        :param atlan: name of the attribute in the metastore
+        :param elastic: name of the field in the search index
+        """
+        self.atlan_field_name = atlan
+        self.elastic_field_name = elastic
+
+    def exists(self) -> Query:
```

Comment:
I would make this and similar read only methods properties.

Reply:
`exists` is literally the only one that has no parameters — every single other method has at least 1 parameter. I think it would be odd if this one way to get a `Query` was a property while _all_ the others are methods (?)

Reply:
In this class yes, but in all the subclasses you have made the methods that don't have parameters properties so I think it would be inconsistent for this to be a method while all the others are properties in the sub-classes.